orderapi/src/main/java/fi/hel/verkkokauppa/order/api/OrderController.java

```diff
@@ -75,9 +75,10 @@ public class OrderController {
 
     @GetMapping(value = "/order/create", produces = MediaType.APPLICATION_JSON_VALUE)
 	public ResponseEntity<OrderAggregateDto> createOrder(@RequestParam(value = "namespace") String namespace,
-                                                         @RequestParam(value = "user") String user) {
+                                                         @RequestParam(value = "user") String user,
+                                                         @RequestParam(required = false, value = "lastValidPurchaseDateTime") String lastValidPurchaseDateTime) {
         try {
-            Order order = orderService.createByParams(namespace, user);
+            Order order = orderService.createByParams(namespace, user, DateTimeUtil.fromFormattedDateTimeOptionalString(lastValidPurchaseDateTime));
```

Comment:
NPO is thrown here if "lastValidPurchaseDateTime" is null. Add a null check and do the formatting only if the value is provided.

Reply:
Good find - I'll add a check here 👍